app/sagas/selectServer.js

```diff
@@ -36,11 +36,26 @@ const handleSelectServer = function* handleSelectServer({ server, version, fetch
 		if (fetchVersion) {
 			serverInfo = yield getServerInfo({ server });
 		}
-		yield AsyncStorage.setItem('currentServer', server);
-		const userStringified = yield AsyncStorage.getItem(`${ RocketChat.TOKEN_KEY }-${ server }`);
 
-		if (userStringified) {
-			const user = JSON.parse(userStringified);
+		const { serversDB } = database.databases;
+		const servers = yield serversDB.objects('servers');
+
+		serversDB.write(() => {
+			try {
+				// eslint-disable-next-line no-plusplus
+				for (let i = 0; i < servers.length; i++) {
+					serversDB.create('servers', { id: servers[i].id, currentServer: false }, true);
```

Comment:
`serversDB.create('servers', { id: servers[i].id, currentServer: servers[i].id === server }, true);` and then no need to make additional request to DB on line next line: `49`

Reply:
nice